File: testprojects/java/benchmark/accuracy-test-2013-04-17/domain/indirect/allowedfrom/CallInstanceMethodIndirectIndirect_VarVarMethod.java
package domain.indirect.allowedfrom;

import domain.indirect.intermediate.BackgroundService;

public class CallInstanceMethodIndirectIndirect_VarVarMethod {
	private BackgroundService bckg;
	private String test;
	
	public CallInstanceMethodIndirectIndirect_VarVarMethod() {
		bckg = new BackgroundService();
	}

	// Double indirect access via normal method 
	public String case6(){
			test = (String) bckg.serviceTwo.serviceOne.getName();
		return test;
	}
}

File: testresources/csharp/benchmark/Domain/Indirect/ViolatingFrom/CallInstanceMethodIndirectIndirect_VarVarMethod.cs
namespace Domain.Indirect.ViolatingFrom
{
	public class CallInstanceMethodIndirectIndirect_VarVarMethod
	{
		private Domain.Indirect.Intermediate.BackgroundService bckg;

		private string test;

        public void Case6()
        {
            bckg = new Domain.Indirect.Intermediate.BackgroundService();
			bckg.serviceTwo.serviceOne.GetName();
		}
	}
}
